In [1]:
import pandas as pd
import matplotlib.pyplot as plt   
%matplotlib inline
import statsmodels.api as sm
from sklearn.model_selection import train_test_split
import numpy as np

In [2]:
df = pd.read_csv('salary.csv')

In [3]:
df = df.dropna(how='any',axis=0)

In [4]:
X = df[['salary', 'exprior', 'yearsrank','market', 'degree', 'otherqual', 'position', 'male', 'Field','yearsabs' ]]
y = df['yearsworked']

In [5]:
X = sm.add_constant(X)

/home/umuzi/.local/lib/python3.6/site-packages/numpy/core/fromnumeric.py:2389: FutureWarning: Method .ptp is deprecated and will be removed in a future version. Use numpy.ptp instead.
  return ptp(axis=axis, out=out, **kwargs)


In [6]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3)

In [7]:
model = sm.OLS(y_train,X_train).fit()

In [8]:
predictions = model.predict(X_train)

In [9]:
print_model = model.summary()
print(print_model)

                            OLS Regression Results                            
Dep. Variable:            yearsworked   R-squared:                       0.906
Model:                            OLS   Adj. R-squared:                  0.904
Method:                 Least Squares   F-statistic:                     336.6
Date:                Wed, 27 Mar 2019   Prob (F-statistic):          3.27e-172
Time:                        09:31:54   Log-Likelihood:                -893.96
No. Observations:                 359   AIC:                             1810.
Df Residuals:                     348   BIC:                             1853.
Df Model:                          10                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.1277      1.595      0.080      0.9

### Does the model significantly predicts the dependent variable? Report the amount of variance explained (R^2) and significance value (p) to support your answer.

The model significantly predicts the dependent variable. This is shown from variance (R^2) 0.967(higher value generally indicates a better fit) with a p value of 0.106(less than 0.05 is considered to be statistically significant)


### What percentage of the variance in employees’ salaries is accounted for by the number of years they have worked?
90.6%



### What does the unstandardized coefficient (B or 'coef' in statsmodels) tell you about the relationship between Years Worked and Salary?
The coeffecient is -3.787e-05 this means that when salary increases by 1, the number of years worked increases by -3.787e-05

### What do the 95% confidence intervals [0.025, 0.975] mean?

### Calculate the expected salary for someone with 12 years’ work experience.

### Calculate the expected salary for someone with 80 years’ work experience. Are there any problems with this prediction? If so, what are they?

### We have only looked at the number of years an employee has worked. What other employee characteristics might influence their salary?

### How does your model compare when running it on the test set - what is the difference in the Root Mean Square Error (RMSE) between the training and test sets?

In [10]:
test_model = sm.OLS(y_test,X_test).fit()

In [11]:
print_test_model = test_model.summary()
print(print_test_model)

                            OLS Regression Results                            
Dep. Variable:            yearsworked   R-squared:                       0.911
Model:                            OLS   Adj. R-squared:                  0.905
Method:                 Least Squares   F-statistic:                     146.7
Date:                Wed, 27 Mar 2019   Prob (F-statistic):           5.72e-70
Time:                        09:31:54   Log-Likelihood:                -373.77
No. Observations:                 154   AIC:                             769.5
Df Residuals:                     143   BIC:                             802.9
Df Model:                          10                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         -1.7179      2.448     -0.702      0.4